### PR TITLE
Task/priority bucket

### LINKED ISSUE #873: Manager - Priority Bucket on Applicant List
# Description
While we've added the ability to mark applicants as "priorities" this hasn't had any effect on the way managers see applicants during screening. Basically, priority applicants need to appear in their own bucket within the first section during applicant screening.

- [ ] Add priority applicant bucket
- [ ] Remove priority applicants from "Vets & Canadian Citizens"
- [ ] Remove Priority Alert above
- [ ] This task requires a unit test.

# Wireframes
image

### PR DESCRIPTION
**Pull request by yonikid15**
Resolves #873 

### Notes:
- Added in priority bucket (accordian) for applicants with "is_priority" value set to TRUE. 

- To test, FIRST create new mock data  `docker exec -t {container name} php artisan db:seed` 
- Go to manager portal and hit "Review Applicants" button on job poster.
- Applicants with "is_priority" set to TRUE should show up in new "Priority Applicants" bucket (unless they do not meet the essential criteria).

### CODE REVIEW

File: resources/assets/js/components/ApplicationReview/ReviewCategory.tsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ const ReviewCategory: React.StatelessComponent<ReviewCategoryProps> = ({
       )
     },
     {
-      title: "Don't Meet Essential Criteria",
+      title: "Don't Need Essential Criteria",
```
Comment (gggrant) [inline resources/assets/js/components/ApplicationReview/ReviewCategory.tsx]: `Meet` here

**Comment by yonikid15** [comment]
Note: Add a icon to represent applicants with priority

File: database/seeds/DevSeeder.php [hunk 2]
```diff
@@ -63,13 +63,27 @@ public function run() : void
             $job->job_applications()->saveMany(factory(JobApplication::class, 5))->create([
                 'job_poster_id' => $job->id
             ]);
+            //Then create one application with a priority user
+            $job->job_applications()->save(factory(JobApplication::class)->create([
+                'job_poster_id' => $job->id,
+                'applicant_id' => factory(Applicant::class)->create([
+                        'user_id' => factory(User::class)->state('priority')->create()->id
```
Comment (Khristophor) [inline database/seeds/DevSeeder.php]: Whenever a state is declared, it's `state()`, whenever a state is used, it's `states()` (thanks Laravel).

https://laravel.com/docs/5.8/database-testing#factory-states

File: resources/assets/js/components/ApplicationReview/ReviewApplications.tsx [hunk 3]
```diff
@@ -80,7 +80,7 @@ const ReviewApplications: React.StatelessComponent<ReviewApplicationsProps> = ({
         </div>
       </div>
 
-      <div className="priority-alert">
+      {/* <div className="priority-alert">
```
Comment (Khristophor) [inline resources/assets/js/components/ApplicationReview/ReviewApplications.tsx]: Can we remove anything we're no longer using? Assuming we don't plan on adding this back in anytime soon?

Comment (yonikid15) [reply resources/assets/js/components/ApplicationReview/ReviewApplications.tsx]: Forgot to remove that.

File: database/seeds/DevSeeder.php (hunk 2 above)
Comment (yonikid15) [reply database/seeds/DevSeeder.php]: I'm not to sure about the syntax, Tryst helped me out with this haha. So what needs to be changed?

File: resources/assets/js/components/ApplicationReview/ApplicationReview.tsx [hunk 4]
```diff
@@ -286,6 +286,17 @@ class ApplicationReview extends React.Component<
...
             </a>
+            {/* This span only shown for priority applicants */}
+            {application.applicant.user.is_priority && (
+              <span className="priority-status">
+                <i
+                  aria-hidden="true"
+                  className="fab fa-product-hunt"
+                  title="The Talent Cloud priority icon"
+                />
+                <span>Priority</span>
```
Comment (Khristophor) [inline resources/assets/js/components/ApplicationReview/ApplicationReview.tsx]: Little late on this, but I guess this should be localized now that we merged in the react-intl branch?

**Pull request merged**